### PR TITLE
=per #15889 Add tck test for save snapshot with same seqNr

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #15889

**Comment by patriknw** [comment]
the test failure was because the snap meta didn't include timestamp in its sorting, fixed

File: akka-persistence/src/main/scala/akka/persistence/SnapshotProtocol.scala [hunk 1]
```diff
@@ -16,6 +16,16 @@ package akka.persistence
...
 //#snapshot-metadata
 
+object SnapshotMetadata {
+  implicit val ordering: Ordering[SnapshotMetadata] = Ordering.fromLessThan[SnapshotMetadata] { (a, b) ⇒
+    if (a eq b) false
+    else if (a.persistenceId != b.persistenceId) a.persistenceId.compareTo(b.persistenceId) < 0
+    else if (a.sequenceNr != b.sequenceNr) a.sequenceNr < b.sequenceNr
+    else if (a.timestamp != b.timestamp) a.timestamp < b.timestamp
+    else false
+  }
```
Comment (ktoso) [inline akka-persistence/src/main/scala/akka/persistence/SnapshotProtocol.scala]: :+1:

File: akka-persistence-tck/src/main/scala/akka/persistence/snapshot/SnapshotStoreSpec.scala [hunk 2]
```diff
@@ -106,5 +106,16 @@ abstract class SnapshotStoreSpec(config: Config) extends PluginSpec(config) {
...
+      val md = metadata(4)
+      snapshotStore.tell(SaveSnapshot(md, s"s-5-modified"), senderProbe.ref)
+      val md2 = senderProbe.expectMsgPF() { case SaveSnapshotSuccess(md2) ⇒ md2 }
+      md2.sequenceNr should be(md.sequenceNr)
+      snapshotStore.tell(LoadSnapshot(pid, SnapshotSelectionCriteria(md.sequenceNr), Long.MaxValue), senderProbe.ref)
+      val result = senderProbe.expectMsgType[LoadSnapshotResult]
+      result.snapshot.get.snapshot should be("s-5-modified")
+      result.snapshot.get.metadata.sequenceNr should be(md.sequenceNr)
+      // metadata timestamp may have been changed
+    }
```
Comment (ktoso) [inline akka-persistence-tck/src/main/scala/akka/persistence/snapshot/SnapshotStoreSpec.scala]: I'm missing the `overwrite` part somehow... what am I forgetting?

Comment (patriknw) [inline akka-persistence-tck/src/main/scala/akka/persistence/snapshot/SnapshotStoreSpec.scala]: `should be("s-5-modified")` verifies that the snapshot is loaded, previous was `"s-5"`

Comment (patriknw) [inline akka-persistence-tck/src/main/scala/akka/persistence/snapshot/SnapshotStoreSpec.scala]: we don't overwrite in the file based store, but semantically a save followed by a load "looks like" an overwrite

Comment (ktoso) [inline akka-persistence-tck/src/main/scala/akka/persistence/snapshot/SnapshotStoreSpec.scala]: Ok, now I know what I missed, this:

```scala
def writeSnapshots(): Seq[SnapshotMetadata] = {
    1 to 5 map { i ⇒
      val metadata = SnapshotMetadata(pid, i + 10)
      snapshotStore.tell(SaveSnapshot(metadata, s"s-${i}"), senderProbe.ref)
```

LGTM then :)

**Comment by ktoso** [comment]
LGTM, please merge :)

**Pull request merged**